### PR TITLE
Update documentation links in README

### LINKED ISSUE #2523: Dead link in menu of arrow-kt.io
#### Steps to reproduce
1. Go to https://arrow-kt.io/docs/typeclasses/intro/
2. On the left, click on "Docs"

#### Actual result
* Link leads to https://arrow-kt.io/docs/apidocs/arrow-core/ which doesn't exist (404)

#### Expected result
* Link leads to an existing location

-----

Side note: I couldn't find separate repository for the website at https://github.com/arrow-kt, I assume that https://github.com/arrow-kt/arrow/issues is the correct place for reporting issues with the website. Please correct me if I'm wrong.

### PR DESCRIPTION
**Pull request by fo2rist**
- remove old not existing links
- update link to newer not deprecated Fx
- add link to Core library
Also add indentation for sub-links within documentation site

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,9 +18,9 @@ blocks to empower users to write pure FP apps and libraries built atop higher or
 Use the list below to learn more about Λrrow's main features.
 
 - [Documentation](http://arrow-kt.io)
-- [Type classes](http://arrow-kt.io/docs/typeclasses/intro/): defining behaviors for data
-- [Effects](http://arrow-kt.io/docs/effects/io/): interfacing with external systems
-- [Optics](http://arrow-kt.io/docs/optics/iso/): inspecting and modifying data structures
+  - [Core](https://arrow-kt.io/docs/core/): error handling and monads
+  - [Fx](https://arrow-kt.io/docs/fx/): interfacing with external systems
+  - [Optics](http://arrow-kt.io/docs/optics/iso/): inspecting and modifying data structures
```
Comment (nomisRev) [inline README.md:23]:
```suggestion
  - [Optics](http://arrow-kt.io/docs/optics/dsl/): inspecting and modifying data structures
```

**Review by nomisRev — APPROVED** [review]
This looks good to me! Thank you for taking the time for this fix @fo2rist 🙏

Small suggestion for Optics, I think the "Overview" page is a much more interesting landing page for Optics.

**Comment by rybak** [comment]
> Closes #2523

This is most likely wrong. See [comment there](https://github.com/arrow-kt/arrow/issues/2523#issuecomment-1032571510) for more details.

**Comment by nomisRev** [comment]
@ryback I see.
Thanks for the update, I'm not so familiar with the website. I've only been writing documentation as markdown files, or inside the actual Kotin code.

I think this fix is still valid, thank you for closing your issue report.

File: README.md (hunk 1 above)
Comment (fo2rist) [reply README.md:23]: Let's do the main Optics page then? `https://arrow-kt.io/docs/optics/`
What do you think @nomisRev ?

**Review by raulraja — APPROVED** [review]
thanks @fo2rist !

**Pull request merged**